Source/WebCore/css/CSSCounterStyleDescriptors.cpp

```diff
@@ -61,99 +62,118 @@ static CSSCounterStyleDescriptors::Ranges translateRangeFromStyleProperties(cons
+{
+    auto value = properties.getPropertyCSSValue(CSSPropertyAdditiveSymbols);
+    if (!value)
+        return { };
+
+    auto& additiveSymbols = downcast<CSSValueList>(*value);
+    CSSCounterStyleDescriptors::AdditiveSymbols result;
+    for (auto& additiveSymbol : additiveSymbols) {
+        if (auto* pair = dynamicDowncast<CSSValuePair>(&additiveSymbol)) {
+            auto weight = downcast<CSSPrimitiveValue>(pair->first()).value<unsigned>();
+            auto symbol = symbolToString(&pair->second());
+            result.constructAndAppend(symbol, weight);
+        } else
+            return { };
+    }
+    return result;
+}
+
 static CSSCounterStyleDescriptors::Pad translatePadFromStyleProperties(const StyleProperties& properties)
 {
-    auto pad = properties.getPropertyCSSValue(CSSPropertyPad).get();
-    if (!pad || !is<CSSValueList>(pad))
-        return CSSCounterStyleDescriptors::Pad();
+    auto value = properties.getPropertyCSSValue(CSSPropertyPad);
+    if (!value)
+        return { };
 
-    auto& list = downcast<CSSValueList>(*pad);
+    auto& list = downcast<CSSValueList>(*value);
     ASSERT(list.size() == 2);
-    auto length = downcast<CSSPrimitiveValue>(list.item(0));
-    if (!length)
-        return CSSCounterStyleDescriptors::Pad();
-    auto lengthValue = length->intValue();
-    ASSERT(lengthValue >= 0);
-    return { static_cast<unsigned>(std::max(0, lengthValue)), symbolToString(list.item(1)) };
+    auto length = downcast<CSSPrimitiveValue>(list[0]).intValue();
+    ASSERT(length >= 0);
+    return { static_cast<unsigned>(std::max(0, length)), symbolToString(&list[1]) };
 }
 
 static CSSCounterStyleDescriptors::NegativeSymbols translateNegativeSymbolsFromStyleProperties(const StyleProperties& properties)
 {
-    auto negative = properties.getPropertyCSSValue(CSSPropertyNegative).get();
-    if (!negative || !is<CSSValueList>(negative))
-        return CSSCounterStyleDescriptors::NegativeSymbols();
+    auto negative = properties.getPropertyCSSValue(CSSPropertyNegative);
+    if (!negative)
+        return { };
 
-    auto& list = downcast<CSSValueList>(*negative);
-    ASSERT(list.size() == 1 || list.size() == 2);
     CSSCounterStyleDescriptors::NegativeSymbols result;
-    if (list.size() >= 1)
-        result.m_prefix = symbolToString(list.item(0));
-    else if (list.size() == 2)
-        result.m_suffix = symbolToString(list.item(1));
-    else {
-        ASSERT_NOT_REACHED();
-        return CSSCounterStyleDescriptors::NegativeSymbols();
+    if (auto list = dynamicDowncast<CSSValueList>(negative.get())) {
+        if (list->size() >= 1)
+            result.m_prefix = symbolToString(list->item(0));
+        else if (list->size() == 2)
+            result.m_suffix = symbolToString(list->item(1));
+        else {
+            ASSERT_NOT_REACHED();
+            return { };
+        return result;
+        }
```

Comment:
The code here looks wrong. From what I can understand, you can only have a list of size 2 or a single value.
```suggestion
    if (auto list = dynamicDowncast<CSSValueList>(negative.get())) {
        ASSERT(list->size() == 2);
        result.m_prefix = symbolToString(list->item(0));
        result.m_suffix = symbolToString(list->item(1));
        return result;
     }
```

Reply:
ops, you are right, thanks